src/reconcile/rec_visibility.c

```diff
@@ -718,6 +718,16 @@ __rec_fill_tw_from_upd_select(
     else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
         WT_ASSERT_ALWAYS(
           session, tombstone != NULL, "The only contents of the update list is a single tombstone");
+
+        /*
+         * The fixed-length column-store implicitly fills the gap with empty records of single
+         * tombstones. We are done with update selection if there is no on-disk entry.
+         */
+        if (vpack == NULL && S2BT(session)->type == BTREE_COL_FIX) {
+            upd_select->upd = tombstone;
+            return (0);
+        }
+
         WT_ASSERT_ALWAYS(session, vpack != NULL && vpack->type != WT_CELL_DEL && !vpack->tw.prepare,
           "No ondisk values found that are not prepared updates");
```

Comment:
@kommiharibabu, should we always mention "prepared updates" in the message? It seems that we can hit this scenario without dealing with prepared updates. This was suggested [here](https://jira.mongodb.org/browse/WT-11113?focusedCommentId=5457662&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-5457662).

Reply:
Thanks. The expectation is that the on-disk value that is not a prepared update when a non-globally visible tombstone is selected. In this scenario, we didn't find any on-disk value. Splitting that check into two can work.

Reply:
Split into two for more clarity on the assert error.